project/FileUtils.scala

```diff
@@ -19,4 +19,30 @@ object FileUtils {
     }
   }
 
+  /** checks if the given jar contains the given entry */
+  def jarContainsEntry(jar: File, entry: String, onlyAtRootLevel: Boolean = false): Boolean = {
+    val zipFs = FileSystems.newFileSystem(jar.toPath)
+    val result = zipFs.getRootDirectories.asScala.exists { zipRootDir =>
+      val iterable =
+        if (onlyAtRootLevel) Files.list(zipRootDir)
+        else Files.walk(zipRootDir)
+      iterable.iterator.asScala.exists(_.toString == "/module-info.class")
```

Comment:
```suggestion
      iterable.iterator.asScala.exists(_.getFileName == entry)
```

`onlyAtRootLevel=false` would break with `toString`. (Or we just remove the flag, since the code path is not used?)

Reply:
Not quite - if we use 'getFileName' we cannot delete files only from select directories. 

Your suggestion as-is would not delete anything, because the filename is actually `module-info.class` (without the `/`). 
And if we use your suggestion and delete `module-info.class` then we'd touch many more jars, because they contain e.g. a `META-INF/9/module-info.class` which doesn't actually harm. And given that this is quite hacky I'd rather keep the changes as small as possible.

Reply:
then the `onlyAtRootLevel` code path is incredibly confusing, and should be deleted on those grounds :)